readme.md

```diff
@@ -9,38 +9,75 @@ Features
 - Different output streams for .js, .d.ts files.
 - Support for sourcemaps using gulp-sourcemaps
 - Compile once, and filter different targets
-- Not just a wrapper around the ```tsc``` command, but a plugin that uses the TypeScript API.
 
 How to install
 --------------
+##### 1. Install Gulp
+```shell
+npm install --global gulp
+```
+##### 2. Install gulp in the project dependency
+```shell
+npm install gulp
+```
+##### 3. Install gulp-typeScript
 ```shell
 npm install gulp-typescript
 ```
 
-Easy usage
+Options
+-------
+- ```out``` (string) - Generate one javascript and one definition file. Only works when no module system is used.
+- ```outDir``` (string) - Move output to a different (virtual) directory. Note that you still need `gulp.dest` to write output to disk.
+- ```removeComments``` (boolean) - Do not emit comments to output.
+- ```noImplicitAny``` (boolean) - Warn on expressions and declarations with an implied 'any' type.
+- ```noLib``` (boolean) - Don't include the default lib (with definitions for - Array, Date etc)
+- ```noEmitOnError``` (boolean) - Do not emit outputs if any type checking errors were reported.
+- ```target``` (string) - Specify ECMAScript target version: 'ES3' (default), 'ES5' or 'ES6'.
+- ```module``` (string) - Specify module code generation: 'commonjs' or 'amd'.
+- ```sourceRoot``` (string) - Specifies the location where debugger should locate TypeScript files instead of source locations.
+- ```declarationFiles``` (boolean) - Generates corresponding .d.ts files.
+- ```noExternalResolve``` (boolean) - Do not resolve files that are not in the input. Explanation below.
+- ```sortOutput``` (boolean) - Sort output files. Usefull if you want to concatenate files (see below).
+- ```typescript``` (object) - Use a different version / fork of TypeScript (see below). Use it like: `typescript: require('typescript')` or `typescript: require('my-fork-of-typescript')`
+
+You can use almost all other options that TypeScript supports too. Only these two are not supported:
+- `sourceRoot` - Use `sourceRoot` option of `gulp-sourcemaps` instead.
+- `rootDir` - Use `base` option of `gulp.src()` instead.
+
+Basis Usage
 ----------
+Below is a minimalist `gulpfile.js` which will compile all TypeScript file in folder `src` and emit a single output file called `out.js` in  `built/local`. To invoke, simple run `gulp`.
+
 ```javascript
-var ts = require('gulp-typescript');
-[...]
-var tsResult = [...].pipe(ts(options));
-tsResult.dts.pipe(...)
-tsResult.js.pipe(...)
+var gulp = require("gulp");
+var ts = require("gulp-typescript");
+
+gulp.task("default", function () {
+  var tsResult = gulp.src("src/*.ts")
+    .pipe(ts({
+        noImplicitAny: true,
+        out: "output.js"
+      }));
+  return tsResult.js.pipe(gulp.dest('built/local'));
+});
 ```
-Example gulpfile:
+Another example of `gulpfile.js`. Instead of creating default task, the file specifies custom named task. To invoke, run `gulp script` instead of `gulp`.
```

Comment:
I'd explain the difference between the previous task and this task, eg that this tasks also generates definition files.